feat: Remove local kubeconfig file creation and `update_kubeconfig` input var

slok opened the pull request:
Fixes #601 

This PR removes the creation of a local kubeconfig file and adds instructions so, users select how it wants to manage the kubeconfig information data.

Also removes the `update_kubeconfig` flag as isn't required anymore.

Note: I don't know where to add the changelog changes for unreleased ones.

devoncrouse commented:
Thanks for submitting the PR - this looks good to me. Can I please ask you to sign the Oracle Contributor Agreement referenced in the earlier comment? Once verified, we'll be able to accept and merge this.

> Note: I don't know where to add the changelog changes for unreleased ones.

It looks like we haven't kept up with this lately. If you'd like to include an entry, you can add an unreleased section with placeholder version, to be associated with a release when next updated like in [this commit](https://github.com/oracle-terraform-modules/terraform-oci-vcn/commit/52158fbc20599bd8027e7b0c9551ac091b2de2a9#diff-bc34fd17a77bda3e2cbef0e783a3ee58a970a9bb65f9476cfdd27c8e78831a93).

slok commented:
I signed yesterday (almost took me more time than the PR itself xD), and it's under review right now.

l also just added all the missing changelog entries (with the information from GitHub releases) and added this PR entry as unreleased.

hyder commented:
@slok: LGTM. We can merge once the OCA is signed and approved.

slok commented:
@devoncrouse @hyder All set! Thanks :)

hyder reviewed (approved):
Approved. Thank you for your contribution
